judge/views/problem.py

```diff
@@ -219,9 +219,70 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.object.can_vote(user)
+        # The vote this user has already cast on this problem.
+        if context['can_vote']:
+            try:
+                vote = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
+            except ObjectDoesNotExist:
+                vote = None
+            context['has_voted'] = vote is not None
+        else:
+            context['has_voted'] = False
+
+        if context['has_voted']:
+            context['voted_points'] = vote.points  # The previous vote's points.
```

Comment:
This is still very scary...`vote` only sometimes exists. Just do something like
```py

context['can_vote'] = self.object.can_vote(user)
if context['can_vote']:
    try:
        context['vote'] = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
    except ObjectDoesNotExist:
        context['vote'] = None
else:
    context['vote'] = None
```
Instead of `voted_points`, use `vote.points`. Instead of `voted_note`, use `vote.note`. Instead of `has_voted`, just check that `vote is not null`.

Reply:
So we remove `voted_note` context and `voted_points` context?

Reply:
Yes